yt_dlp/YoutubeDL.py

```diff
@@ -2230,7 +2245,7 @@ def restore_last_token(self):
         return _build_selector_function(parsed_selector)
 
     def _calc_headers(self, info_dict):
-        res = merge_headers(self.params['http_headers'], info_dict.get('http_headers') or {})
+        res = CaseInsensitiveDict(self.params['http_headers'], info_dict.get('http_headers') or {})
```

Comment:
It looks like what we actually need is a `CaseInsensitiveChainMap` instead of copying and updating dicts all the time

Reply:
I'm not really sure how it would help in this situation?

Reply:
In init, `self.params['http_headers'] = std_headers.new_child(self.params.get('http_headers'))`
here, `res = self.params['http_headers'].new_child(info_dict.get('http_headers'))`